docs/self-managed/platform-deployment/kubernetes-helm.md

```diff
@@ -137,3 +137,73 @@ If the output of the `describe` command was not beneficial, tail the logs of the
+
+However, if Identity is enabled (this is the default), the upgrade path is a bit more complex than just running `helm upgrade`.
+
+Read the next sections to familiarize yourself with the upgrade process.
+
+### Upgrading with Identity and Secrets
+
+If you have installed the Camunda Platform 8 Helm charts before with default values, this means Identity and the related authentication mechanism are enabled. For authentication, the Helm charts generate for each web app the secrets randomly if not specified on installation. If you just run `helm upgrade` to upgrade to a newer chart version, you likely will see the following return:
+
+```shell
+$ helm upgrade camunda-platform-test camunda/camunda-platform 
+Error: UPGRADE FAILED: execution error at (camunda-platform/charts/identity/templates/tasklist-secret.yaml:10:22): 
+PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
+                 Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
+                 Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
+
+    'global.identity.auth.tasklist.existingSecret' must not be empty, please add '--set global.identity.auth.tasklist.existingSecret=$TASKLIST_SECRET' to the command. To get the current value:
+
+        export TASKLIST_SECRET=$(kubectl get secret --namespace "zell-c8-optimize" "camunda-platform-test-tasklist-identity-secret" -o jsonpath="{.data.tasklist-secret}" | base64 --decode)
+```
+
+As described earlier per default, this occurs because Secrets are randomly generated on the first Helm installation if not further specified. We use a library chart [provided by Bitnami](https://github.com/bitnami/charts/tree/master/bitnami/common) for this. The generated Secrets are persisted on persistent volume claims (PVCs), which are not maintained by Helm.
+
+If you remove the Helm chart release or do an upgrade, PVCs are not removed nor recreated. On an upgrade, Secrets can be recreated by Helm, and could lead to the regeneration of the Secret values. This would mean that newly-generated Secrets wouldn't longer match with the persisted secrets. To avoid such an issue, Bitnami blocks the upgrade path and prints the help message as shown above.
+
+In the error message, Bitnami links to their well-described [troubleshooting guide](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases). However, to avoid confusion we will step through the troubleshooting process in this guide as well.
+
+### Secrets extraction
+
+For a successful upgrade, you first need to extract all Secrets which were previously generated.
+
+:::note
+You also need to extract all Secrets which were generated for Keycloak, since Keycloak is a dependency of Identity.
```

Comment:
@Zelldon Is this note what you mean? I adjusted it to clarify.

Reply:
Looks great :+1: Wasn't aware of the `:::note` Thanks :+1: